### PR TITLE
modernize cloud code

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

File: integration/cloud/main.js [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
-Parse.Cloud.job('CloudJob2', function(request, response) {
-  setTimeout(function() {
-    response.success({
-      status: 'cloud job completed'
-    })
-  }, 3000);
+Parse.Cloud.job('CloudJob2', function() {
+  return new Promise((resolve) => {
+    setTimeout(function() {
+      resolve({
```
Comment (dplewis) [inline integration/cloud/main.js]: Nice

Comment (flovilmart) [reply integration/cloud/main.js]: I hope it's gonna work :)

**Comment by dplewis** [comment]
I've been running into this test failing a lot.

```
1) Parse Query can do a query with a limit
  Message:
[31m    Failed: 'qux' == 'baz'[0m
  Stack:
    error properties: Object({ generatedMessage: true, code: 'ERR_ASSERTION', actual: 'qux', expected: 'baz', operator: '==' })
        at <Jasmine>
        at Parse.Object.saveAll.then.then (/Users/Squad/Github/Parse-SDK-JS/integration/test/ParseQueryTest.js:54:14)
        at <Jasmine>
        at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

**Comment by flovilmart** [comment]
@dplewis it makes sense as there is o guaraneed order when you save.

**Pull request merged**